### PR TITLE
Update documentation for custom widget

### LINKED ISSUE #2581: Update documentation for custom ui components
**Describe the Issue**

**Would you like to work on the issue?**

### PR DESCRIPTION
**Pull request by MJ1998**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #2581 

**Description**
Clear and concise code change description.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**

- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 1]
```diff
@@ -79,53 +79,65 @@ component.
...
-    ([example](https://github.com/google/android-fhir/blob/master/catalog/src/main/java/com/google/android/fhir/catalog/CustomQuestionnaireFragment.kt#L22)).
-    It should return a list that contains your
-    `QuestionnaireItemViewHolderFactoryMatcher`.
-3. When rendering your questionnaire, use your custom implementation of
-    <code>QuestionnaireFragment</code> instead.
+This guide outlines the step-by-step process to create and incorporate custom widgets into FHIR questionnaires using the Android FHIR SDK.
+
+Note: Examples given below are all from the [catalog](https://github.com/google/android-fhir/tree/master/catalog) app in this repository. We recommend you try out the application to see these concepts in action.
+
+1. **Create a Layout for the Custom Component:**
```
Comment (jingtang10) [inline docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:86]:
```suggestion
1. Create a layout for the Custom Component
```

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 2]
```diff
@@ -79,53 +79,65 @@ component.
...
+
+   Design a layout file (XML) to define the visual structure and appearance of your custom widget.
```
Comment (jingtang10) [inline docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:88]:
```suggestion
   Design an XML layout file to define the visual structure and appearance of your custom widget.
```

**Review by jingtang10 — APPROVED** [review]
thanks jajoo! this is really great!

mostly some formatting nitpicks re capitalisation and punctuation.

**Review by jingtang10 — CHANGES_REQUESTED** [review]
jajoo making more changes. pls let me know when ready.

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 3]
```diff
@@ -79,53 +79,65 @@ component.
...
+
+    * Example: See this [layout example](https://github.com/google/android-fhir/blob/master/contrib/locationwidget/src/main/res/layout/location_widget_view.xml).
+
+2. **Create a `QuestionnaireItemViewHolderFactory` Class:**
```
[the remark this answers is not in the document]
Comment (MJ1998) [reply docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:92]: Them in bold look better.

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 4]
```diff
@@ -79,53 +79,65 @@ component.
...
+   For each custom `ViewHolderFactory`, create a corresponding `QuestionnaireItemViewHolderFactoryMatcher` object.
+
+    * Example: Refer to these [matcher examples](https://github.com/google/android-fhir/blob/aea1af318b327716db3a1f6228105de5c74b78e3/datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt#L538).
+
+   Each matcher should define:
+
+    * `factory`: The custom `ViewHolderFactory` instance.
+    * `matches`: A predicate function that takes a [`Questionnaire.QuestionnaireItemComponent`](https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-structures-r4/org/hl7/fhir/r4/model/Questionnaire.QuestionnaireItemComponent.html) and returns `true` if this factory is suitable for rendering that item.
+
+4. **Create a `QuestionnaireItemViewHolderFactoryMatchersProviderFactory`:**
```
[the remark this answers is not in the document]
Comment (MJ1998) [reply docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:118]: Them in bold look better.

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 5]
```diff
@@ -79,53 +79,65 @@ component.
...
+
+    * Example: This [example factory class](https://github.com/google/android-fhir/blob/master/catalog/src/main/java/com/google/android/fhir/catalog/ContribQuestionnaireItemViewHolderFactoryMatchersProviderFactory.kt) demonstrates this.
+
+5. **Register the Factory in `DataCaptureConfig`:**
+
+   In your `DataCaptureConfig` object, set the `questionnaireItemViewHolderFactoryMatchersProviderFactory` property to your custom factory instance.
+
+    * Example: See how this is done in [this code](https://github.com/google/android-fhir/blob/ced8527a5481972591615ad4364487e89130fb6e/catalog/src/main/java/com/google/android/fhir/catalog/CatalogApplication.kt#L42C5-L47C8).
+
+6. **Use the Custom Widget in Your Questionnaire Fragment:**
```
[the remark this answers is not in the document]
Comment (MJ1998) [reply docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:130]: Them in bold look better.

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 6]
```diff
@@ -79,53 +79,65 @@ component.
...
+    * `factory`: The custom `ViewHolderFactory` instance.
+    * `matches`: A predicate function that takes a [`Questionnaire.QuestionnaireItemComponent`](https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-structures-r4/org/hl7/fhir/r4/model/Questionnaire.QuestionnaireItemComponent.html) and returns `true` if this factory is suitable for rendering that item.
+
+4. **Create a `QuestionnaireItemViewHolderFactoryMatchersProviderFactory`:**
+
+   Create a factory class that implements this interface. It acts as a central registry for your custom widget associations.
+
+    * Example: This [example factory class](https://github.com/google/android-fhir/blob/master/catalog/src/main/java/com/google/android/fhir/catalog/ContribQuestionnaireItemViewHolderFactoryMatchersProviderFactory.kt) demonstrates this.
+
+5. **Register the Factory in `DataCaptureConfig`:**
```
[the remark this answers is not in the document]
Comment (MJ1998) [reply docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:124]: Them in bold look better.

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**